core/src/main/java/io/questdb/griffin/engine/groupby/GroupByUtils.java

```diff
@@ -538,4 +540,80 @@ private static Function createColumnFunction(
         }
         return func;
     }
+
+    /**
+     * Checks if an alias appears in the columns and function args of a model.
+     * <p>
+     * Consider a query:<br>
+     * (select x1, sum(x1) from (select x as x1 from y))<br>
+     * If you were to move lift the column from the RHS to the LHS, you might get:<br>
+     * (select x x1, sum(x1) from y)<br>
+     * Now x1 is invalid.
+     * @param model
+     * @param alias
+     * @return
+     */
+    public static boolean aliasAppearsInColsAndFuncArgs(QueryModel model, CharSequence alias) {
```

Comment:
It might be a better idea to keep this method in `SqlOptimiser`. We already have a stack-based framework for tree traversal. See `SqlOptimiser#traversalAlgo` and how it's used. An even simpler option would be to use `sqlNodeStack` field in `SqlOptimiser`, see `hasAggregates()` method as an example.